model/oauth.go

```diff
@@ -100,6 +101,10 @@ func (a *OAuthApp) IsValid() *AppError {
 		}
 	}
 
+	if err := a.Scopes.Validate(); err != nil {
+		return NewAppError("OAuthApp.IsValid", "model.oauth.is_valid.scopes.app_error", nil, err.Error(), http.StatusBadRequest)
```

Comment:
Possibly use `http.StatusUnauthorized` instead?

```suggestion
		return NewAppError("OAuthApp.IsValid", "model.oauth.is_valid.scopes.app_error", nil, err.Error(), http.StatusUnauthorized)
```

Reply:
Form the context, it might be better to distinguish between bad request errors due to data validation?

Reply:
This means that the payload of the request is not valid, how is it a `StatusUnauthorized`?